ublox_cli/src/main.rs

```diff
@@ -78,86 +82,123 @@ impl Device {
                 .help("Baud rate of the port"),
         )
+        .arg(
+            Arg::new("stop-bits")
+                .long("stop-bits")
+                .help("Number of stop bits to use")
+                .required(false)
+                .value_parser(["1", "2"])
+                .default_value("1"),
+        )
+        .arg(
+            Arg::new("data-bits")
+                .long("data-bits")
+                .help("Number of data bits to use")
+                .required(false)
+                .value_parser(["5", "6", "7", "8"])
+                .default_value("8"),
+        )
         .get_matches();
 
-    let port = matches.value_of("port").unwrap();
-    let baud: u32 = matches
-        .value_of("baud")
-        .unwrap_or("9600")
-        .parse()
-        .expect("Could not parse baudrate as an integer");
-
-    let s = serialport::SerialPortSettings {
-        baud_rate: baud,
-        data_bits: serialport::DataBits::Eight,
-        flow_control: serialport::FlowControl::None,
-        parity: serialport::Parity::None,
-        stop_bits: serialport::StopBits::One,
-        timeout: Duration::from_millis(1),
+    let port = matches
+        .get_one::<String>("port")
+        .expect("Expected required 'port' cli argumnet");
+    let baud = matches.get_one::<u32>("baud").cloned().unwrap_or(9600);
+    let stop_bits = match matches.get_one::<String>("stop-bits").map(|s| s.as_str()) {
+        Some("2") => SerialStopBits::Two,
+        _ => SerialStopBits::One,
     };
-    let port = serialport::open_with_settings(port, &s).unwrap();
+    let data_bits = match matches.get_one::<String>("data-bits").map(|s| s.as_str()) {
+        Some("5") => SerialDataBits::Five,
+        Some("6") => SerialDataBits::Six,
+        Some("7") => SerialDataBits::Seven,
+        _ => SerialDataBits::Eight,
+    };
+
+    let builder = serialport::new(port, baud)
+        .stop_bits(stop_bits)
+        .data_bits(data_bits)
+        .timeout(Duration::from_millis(1))
+        .parity(SerialParity::None)
+        .flow_control(SerialFlowControl::None);
+
+    println!("{:?}", &builder);
+    let port = builder.open().unwrap_or_else(|e| {
+        eprintln!("Failed to open \"{}\". Error: {}", port, e);
+        ::std::process::exit(1);
+    });
     let mut device = Device::new(port);
 
     // Configure the device to talk UBX
+    println!("Configuring UART1 port ...");
     device
         .write_all(
             &CfgPrtUartBuilder {
                 portid: UartPortId::Uart1,
                 reserved0: 0,
                 tx_ready: 0,
                 mode: UartMode::new(DataBits::Eight, Parity::None, StopBits::One),
-                baud_rate: 9600,
-                in_proto_mask: InProtoMask::all(),
-                out_proto_mask: OutProtoMask::UBLOX,
+                baud_rate: baud,
+                in_proto_mask: InProtoMask::UBLOX,
+                out_proto_mask: OutProtoMask::union(OutProtoMask::NMEA, OutProtoMask::UBLOX),
```

Comment:
These could also benefit command line options in the future

Reply:
Good point, I can add them in a next PR. I have more coming :)